ovirt/resource_ovirt_vm.go

```diff
@@ -884,27 +912,61 @@ func ovirtAttachDisks(s []interface{}, vmID string, meta interface{}) error {
 			return err
 		}
 
-		err = resource.Retry(2*time.Minute, func() *resource.RetryError {
-			addAttachmentResp, err := vmService.DiskAttachmentsService().
-				Add().
-				Attachment(da).
-				Send()
-			if err != nil {
-				return resource.RetryableError(fmt.Errorf("failed to attach disk: %s, wait for next check", err))
-			}
-			_, ok := addAttachmentResp.Attachment()
-			if !ok {
-				return resource.RetryableError(fmt.Errorf("failed to attach disk: not exists in response, wait for next check"))
-			}
-			return nil
-		})
+		attachment, err := attachDisk(vmService.DiskAttachmentsService(), da, attachmentExists)
+		if err != nil {
+			return fmt.Errorf("failed to attach disk: %s", err)
+		}
+		err = conn.WaitForDisk(attachment.MustId(), ovirtsdk4.DISKSTATUS_OK, 20*time.Minute)
 		if err != nil {
 			return err
 		}
+		return nil
 	}
 	return nil
 }
 
+// attachDisk will attach a disk to vm or update an existing one. Returns the
+// new attachment or error.
+func attachDisk(service *ovirtsdk4.DiskAttachmentsService, attachment *ovirtsdk4.DiskAttachment, update bool) (*ovirtsdk4.DiskAttachment, error) {
+	if update {
```

Comment:
Are you sure you want to update/add disk each time?
If the bootable disk exists, the update bool will be true. Therefore automatically you will update the disk.
What if size is not provided and the user want to continue with the existing disk without any change? Will you force an update action that does nothing?

Reply:
this whole section is ignored by simply omitting `block_device` from the declaration.